### PR TITLE
Implementation of Fluids for Forge

### PR DESCRIPTION
**Pull request by gabizou**
[SpongeAPI](https://github.com/SpongePowered/SpongeAPI/pull/1024) | [SpongeCommon](https://github.com/SpongePowered/SpongeCommon/pull/410) | SpongeForge

This adds full compatibility with Forge's Fluids API integration mixing into Forge's `FluidStack` and the `DataProcessor`s to handle `IFluidHandler`s and `IFluidItemContainer`s.

### CODE REVIEW

File: src/main/java/org/spongepowered/mod/mixin/core/forge/fluids/package-info.java [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+@org.spongepowered.api.util.annotation.NonnullByDefault package org.spongepowered.mod.mixin.core.forge.fluids;
```
Comment (kashike) [inline src/main/java/org/spongepowered/mod/mixin/core/forge/fluids/package-info.java]: Missing newline

File: src/main/java/org/spongepowered/mod/mixin/core/forge/fluids/MixinFluidStack.java [hunk 2]
```diff
@@ -0,0 +1,291 @@
...
+    }
+
+    @Override
+    public DataTransactionResult copyFrom(DataHolder that, MergeFunction function) {
+        return offer(that.getContainers(), function);
+    }
+
+    @Override
+    public Collection<DataManipulator<?, ?>> getContainers() {
+        return null;
```
Comment (JBYoshi) [inline src/main/java/org/spongepowered/mod/mixin/core/forge/fluids/MixinFluidStack.java]: Could you at least make this return a non-`null` value?

Comment (gabizou) [inline src/main/java/org/spongepowered/mod/mixin/core/forge/fluids/MixinFluidStack.java]: Note that it's a WIP?

Comment (JBYoshi) [inline src/main/java/org/spongepowered/mod/mixin/core/forge/fluids/MixinFluidStack.java]: Sorry, didn't see that.

**Pull request closed without merging**